views/components/ConfigurationsList/index.jsx

```diff
@@ -0,0 +1,851 @@
+      })
+      .then((response) => {
+        if (response.status == 201) {
+          setState((prevState) => {
+            return {
+              ...prevState,
+              configurations: update(state.configurations, {
+                [state.selectedConfig['index']]: {
+                  ['readers']: {
+                    $set: state.shared.map((o) => {
+                      return o.value
+                    }),
+                  },
+                },
+              }),
+            }
+          })
+        }
+        closeSearchUsers()
+      })
+  }
+  const handleChange = (name) => (value) => {
+    let uid = user.uid
+    let names = value.map((o) => {
+      return o.value
+    })
+    if (names.indexOf(uid) === -1) {
+      throw new Error('Unable to delete owner.')
+    }
+    setState((prevState) => {
+      return { ...prevState, [name]: value }
+    })
+  }
+  const handleChangeFile = (e) => {
+    e.preventDefault()
+    const file = e.target.files ? e.target.files[0] : ''
+    new Response(file)
+      .json()
+      .then(async (json) => {
+        const res = await window.fetch(
+          `${basePath}/api/v1/users/${user.uid}/config/file`,
+          {
+            method: 'POST',
+            headers: {
+              'Content-Type': 'application/json',
+            },
+            credentials: 'same-origin',
+            body: JSON.stringify(json),
+          }
+        )
+        if (res.status === 200) {
+          window.location = `${basePath}/u/configure?u=success`
+        } else if (res.status === 400) {
+          window.location = `${basePath}/u/configure?u=invalid`
+        } else {
+          window.location = `${basePath}/u/configure?u=error`
+        }
+      })
+      .catch((err) => {
+        console.error(err.message)
+      })
+  }
+  const changeDefaultConfig = (e, checked, index) => {
+    updateUserPreferences(index, 'index')
+  }
+
+  const actions = [
+    <Button
+      onClick={closeSearchUsers}
+      style={{
+        color: '#5790bd',
+      }}
+      key="cancel"
+    >
+      Cancel
+    </Button>,
+    <Button
+      variant="outlined"
+      style={{
+        borderColor: '#5790bd',
+        paddingTop: '11px',
+        color: '#ffffff',
+        backgroundColor: '#5790bd',
+        marginLeft: '12px',
+      }}
+      keyboardFocused={true}
+      onClick={shareWithUsers}
+      key="submit"
+    >
+      Submit
+    </Button>,
+  ]
+  const selectStyles = {
+    input: (base) => ({
+      ...base,
+      color: theme.palette.text.primary,
+      '& input': {
+        font: 'inherit',
+      },
+    }),
+  }
+
+  if (state?.configurations?.length <= 0) return <div>Loading...</div>
+
+  return (
+    <div>
+      <GridList
+        style={{
+          padding: '2px',
+          overflowY: 'auto',
+          marginBottom: '128px',
+        }}
+        cols={state.gridCols}
+        cellHeight="auto"
+      >
+        {generateCards(state.configurations, state.preferences)}
```

Comment:
we shouldn't need to pass the state to this function since it already has access, unless we want to split out another component for these cards and pass these in as props.

Reply:
Thanks for this. I feel like what this comment is more suited for the scope of #296